src/libraries/Common/tests/TestUtilities/System/RetryHelper.cs

```diff
@@ -35,20 +39,34 @@ public static void Execute(Action test, int maxAttempts = 5, Func<int, int> back
             var exceptions = new List<Exception>();
             for (int i = 1; i <= maxAttempts; i++)
             {
+                Exception lastException;
                 try
                 {
                     test();
                     return;
                 }
                 catch (Exception e) when (retryWhen(e))
                 {
+                    lastException = e;
                     exceptions.Add(e);
                     if (i == maxAttempts)
                     {
                         throw new AggregateException(exceptions);
                     }
                 }
 
+                string diagnostic = $"RetryHelper: retrying {testName} {i}th time of {maxAttempts}: got {lastException.Message}";
+                if (PlatformDetection.IsInHelix && testName.Contains("FileSystemWatcher_"))
```

Comment:
This class is in the common test utilities, which means the `Contains` call would impact any test invoking this method in a Helix machine.

Maybe it could be better to add an `Execute` method that is exclusive to `FileSystemWatcher` tests and they _always_ log the diagnostic to console.

Also, why do they need to check if we are in Helix? Wouldn't these diagnostics be helpful when debugging locally? Or is there a restriction I'm not aware of?

Reply:
I don't want to pollute the console output for developers who generally have nothing to do with FSW or its tests. The idea is that I can get this if by opting-in, or if I grep the Helix logs.

Note that the code only gets here if there's a retry happening. I couldn't think of a cleaner way to customize it for the FSW tests without duplicating the code, adding yet another parameter, etc.

Reply:
I'll copy the code.